extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -102,9 +121,10 @@ public void initialize() {
         }
 
         DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
-        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        builder.setPskStore(new StaticPskStore(configuration.username, configuration.userkey.getBytes()));
         dtlsConnector = new DTLSConnector(builder.build());
         endPoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+        endPoint.setExecutor(Executors.newSingleThreadScheduledExecutor(new Utils.DaemonThreadFactory()));
```

Comment:
This is an ugly workaround to fight symptoms, which might introduce other issue like resource leaks.
I'd prefer adding proper logging instead as it had been suggested in https://github.com/eclipse/californium/pull/452#issuecomment-339273784.
Btw, please don't mix such things in a single PR, but rather created separate ones for separate issues.

Reply:
> Btw, please don't mix such things in a single PR, but rather created separate ones for separate issues.

OK - point taken

>This is an ugly workaround to fight symptoms, which might introduce other issue like resource leaks.
I'd prefer adding proper logging instead as it had been suggested in eclipse/californium#452 (comment).

The suggestion actually was to use own executor:
>As proposed in eclipse/smarthome#3867 (comment), if you cant wait, I would propose you use
>```
>ScheduledExecutorService executor = 
>   Executors.newSingleThreadScheduledExecutor(new Utils.DaemonThreadFactory());
>CoapEndpoint.setExecutor(executor);
>```
>in the smarthome tradfri binding. This should not shutdown the executor on destroying the CoapEndpoint, because the EndpointObserver is not used.